### PR TITLE
feat(calendar) : #MZI-161 implement cron for calendar requests

### PR DESCRIPTION
**Pull request by alicedraillard**
## Describe your changes
add cron that runs to call calendar worker (which is not implemented yet).
default time is every 1 minute.

new config:
```
"zimbra-ical-cron" : "$zimbraICalCron"
```

## Issue ticket number and link
[MZI-161](https://jira.support-ent.fr/browse/MZI-161)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: zimbra/src/main/java/fr/openent/zimbra/helper/ConfigManager.java [hunk 1]
```diff
@@ -131,6 +133,7 @@ public ConfigManager(JsonObject config) {
         this.sendTimeout = config.getInteger("send-timeout", 5000);
         this.structureToSynchroABLimit = config.getInteger("limit-structures-synchro-ab",5);
         this.filterUserProfileSynchAB = config.getString("filter-profile-sync-ab","");
+        this.zimbraICalCron = config.getString("zimbra-ical-cron", "0 */1 * * * ? *");
```
Comment (alicedraillard) [inline zimbra/src/main/java/fr/openent/zimbra/helper/ConfigManager.java]: default time every 1 minute

Comment (Matiyeu) [reply zimbra/src/main/java/fr/openent/zimbra/helper/ConfigManager.java]: Tu peux direct mettre ça : `0 * * * * ? *`

File: zimbra/src/main/java/fr/openent/zimbra/cron/ICalRequestCron.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+public class ICalRequestCron extends ControllerHelper implements Handler<Long> {
+    private static final Logger log = LoggerFactory.getLogger(ICalRequestCron.class);
+
+    public ICalRequestCron () {
+    }
+
+    @Override
+    public void handle(Long event) {
+        log.info("CRON started");
```
Comment (Matiyeu) [inline zimbra/src/main/java/fr/openent/zimbra/cron/ICalRequestCron.java]:
```suggestion
        log.info("[ZimbraConnector@ICalRequestCron] ICalRequestCron started");
```

File: zimbra/deployment/zimbra/conf.json.template [hunk 3]
```diff
@@ -26,6 +26,7 @@
     "address-book-account" : "$zimbraAddressBookAccount",
     "admin-password" : "$zimbraAdminPassword",
     "zimbra-synchro-cron" : "$zimbraSynchroCron",
+    "zimbra-ical-cron" : "$zimbraICalCron",
```
Comment (Matiyeu) [inline zimbra/deployment/zimbra/conf.json.template:29]: à ajouter dans le readme et conf.j2

**Review by aymericdevriesere — APPROVED** [review]

File: Readme.md [hunk 4]
```diff
@@ -1,4 +1,4 @@
-# About Zimbra Connetor
```
Comment (BastienBenard) [inline Readme.md:1]: Ce serait cool d'avoir une description de ce que fait ta nouvelle config. Un peu comme le readme https://github.com/OPEN-ENT-NG/support-pivot

File: zimbra/src/main/java/fr/openent/zimbra/cron/ICalRequestCron.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+public class ICalRequestCron extends ControllerHelper implements Handler<Long> {
+    private static final Logger log = LoggerFactory.getLogger(ICalRequestCron.class);
+
+    public ICalRequestCron () {
+    }
+
+    @Override
+    public void handle(Long event) {
+        log.info("[ZimbraConnector@ICalRequestCron] ICalRequestCron started");
+        //todo call worker
```
Comment (BastienBenard) [inline zimbra/src/main/java/fr/openent/zimbra/cron/ICalRequestCron.java:17]: ce sera fais plus tard ?

**Review by Matiyeu — APPROVED** [review]

**Review by BastienBenard — APPROVED** [review]

File: zimbra/src/main/java/fr/openent/zimbra/cron/ICalRequestCron.java (hunk 5 above)
Comment (alicedraillard) [reply zimbra/src/main/java/fr/openent/zimbra/cron/ICalRequestCron.java:17]: oui en [MZI-162](https://jira.support-ent.fr/browse/MZI-162)

**Pull request merged**